app/models/manageiq/providers/openshift/container_manager_mixin.rb

```diff
@@ -36,13 +21,17 @@ def openshift_connect(hostname, port, options)
     end
 
     def openshift_v3_connect(hostname, port, options)
-      options = {:path => '/oapi', :version => api_version}.merge(options)
+      options = {:path => '/oapi', :version => "v1"}.merge(options)
       kubernetes_connect(hostname, port, options)
     end
 
     def openshift_v4_connect(hostname, port, options)
-      api_group = options[:api_group] || "apps.openshift.io"
-      options = {:path => "/apis/#{api_group}", :version => api_version}.merge(options)
+      api_group = options[:api_group] || "apps.openshift.io/v1"
+
+      api_path, api_version = api_group.split("/")
+      api_version ||= "v1"
```

Comment:
I think the whole direction makes sense, and ideally every caller that specifies an API group should also specify its version.

Here openshift_v4_connect has different patam names from openshift_v3_connect?
For automatic fall over to work, they must use same interface, no?

Reply:
> and ideally every caller that specifies an API group should also specify its version.

Yeah great idea, done